### ISSUE TITLE
Increase jpg compression?

### ISSUE THREAD

**Issue by larssandergreen** [veda-consulting/uk.co.vedaconsulting.mosaico#417]
I'm not sure if this is a Mosaico issue or a Mosaico CiviCRM issue, but it seems jpg image compression is very minimal, resulting in large images. For example, with a 534px wide image in the Versafix template and the 2x resolution setting, you can easily end up with a 500kB+ image. If you have a few images of that size, your email could end up fairly slow to load and that can affect deliverability. If you had the 3x image size setting enabled, you could have very large images indeed!

Would it be possible to increase the compression for jpgs? Image size could easily be halved or more without noticeably affecting image quality, especially when the image is already 2x or 3x resolution.

**Comment by larssandergreen** [comment]
Looks like the jpeg compression could be set [here](https://github.com/veda-consulting-company/uk.co.vedaconsulting.mosaico/blob/9ff7a27fdfb2ebecd77294a287a1ed057d9cedd8/CRM/Mosaico/Graphics/Intervention.php#L120) and [here](https://github.com/veda-consulting-company/uk.co.vedaconsulting.mosaico/blob/9ff7a27fdfb2ebecd77294a287a1ed057d9cedd8/CRM/Mosaico/Graphics/Intervention.php#L158).

Looks like [default is 90](http://image.intervention.io/api/save), but we could safely decrease that to 82 (this is default for Wordpress, for example).

I propose to add a field to let users set this on the Mosaico settings page, with help text indicating that the default is 82, but the value can be set lower if you are using 2x or 3x images. I'm not sure how much lower; I will do some testing if I can't find any specific resources on this.

I'll provide a patch if there is support for this change.

**Comment by totten** [comment]
I would suggest double-checking the lifecycle of an image. (My recollection is hazy.) Assuming the extension only applies a single pass of scaling/re-compression/lossiness, then yeah - it seems fair to expose some straight-forward backend setting(s) on image-quality.

(However, if the image might go through multiple passes -- e.g. one pass when you first upload, and another pass when you (re)use from a gallery -- then it would merits a little think on the cumulative impact of the quality-setting.)

**Comment by larssandergreen** [comment]
Good point @totten. Looks like the images in the gallery are being compressed to 90 when they are saved (and are resized if they are large as well). It's not clear to me where this is happening though.

So would it make sense to store the gallery images at 100 quality? They are only being stored at 1920x1080 max, I believe (or something similar), but that would still more than double disk space usage, possibly in the range of 1-2MB. Perhaps something higher than 90, but less than 100, would make sense.
